src/clj/puppetlabs/services/file_serving/file_metadata/file_metadata_core.clj

```diff
@@ -0,0 +1,200 @@
+(ns puppetlabs.services.file-serving.file-metadata.file-metadata-core
+  (:import (org.apache.commons.codec.digest DigestUtils)
+           [java.text SimpleDateFormat]
+           [java.util Locale Date])
+  (:require [puppetlabs.puppetserver.ringutils :as ringutils]
+            [puppetlabs.puppetserver.liberator-utils :as utils]
+            [puppetlabs.services.protocols.puppet-fileserver-config :as fileserver-config]
+            [clojure.tools.logging :as log]
+            [clojure.java.io :as io]
+            [clojure.string :as string]
+            [schema.core :as schema]
+            [cheshire.core :as cheshire]
+            [compojure.core :as compojure :refer [GET ANY PUT]]
+            [liberator.core :refer [defresource]]
+            [liberator.representation :as representation]
+            [liberator.dev :as liberator-dev]
+            [puppetlabs.services.file-serving.config.puppet-fileserver-config-core :as config]
+            [puppetlabs.services.file-serving.utils.posix-utils :as posix-utils]
+            )
+  (:use [clojure.java.io :only (input-stream)]
+        [ring.middleware.params]
+        ))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;;; 'handler' functions for HTTP endpoints
+
+(defn- checksum-of-file
+  [file]
+  "Compute the md5 checksum of a given file path"
+  (with-open [r (java.nio.file.Files/newInputStream file (into-array java.nio.file.OpenOption []))]
+    {
+     :type "md5"
+     :value (str "{md5}" (DigestUtils/md5Hex r))
+     }
+  ))
+
+(def date-format (SimpleDateFormat. "yyyy.MM.dd HH:mm:ss Z" Locale/ROOT))
+
+(defn date->ruby
+  [time]
+  (.format date-format (Date. time)))
+
+(defn ruby->date
+  [date]
+  (.parse date-format date))
+
+(defn- checksum-of-dir
+  [posix-attr]
+  "Compute the ctime checksum of a given directory"
+    {
+     :type "ctime"
+     :value (str "{ctime}" (date->ruby (.toMillis (posix-attr "creationTime"))))
+     }
+    )
+
+(def base-metadata
+  {
+    :data          {
+                     }
+    :document_type "FileMetadata"
+    :metadata      {
+                     :api_version 1
+                     }
+   })
```

Comment:
Style/formatting nitpick: in most Clojure code I've seen/written, braces, brackets, and parentheses are never on their own line (so basically, follow https://github.com/bbatsov/clojure-style-guide#gather-trailing-parens for braces and brackets as well).

So, this would look like:
``` clojure
(def base-metadata
  {:data          {}
   :document_type "FileMetadata"
   :metadata      {:api_version 1}})
```